File: 118-pascals-triangle/go/main.go
package main

func generate(numRows int) [][]int {
	if numRows == 0 {
		return [][]int{}
	}

	res := [][]int{{1}}

	for i := 2; i <= numRows; i++ {
		temp := []int{1}
		last := res[len(res)-1]

		for j := 1; j < i-1; j++ {
			temp = append(temp, last[j-1]+last[j])
		}

		temp = append(temp, 1)
		res = append(res, temp)
	}

	return res
}

File: 118-pascals-triangle/cpp/main.cpp
class Solution {
 public:
  vector<vector<int>> generate(int numRows) {
    if (numRows == 0) return {};
    vector<vector<int>> res = {{1}};

    for (int i = 2; i <= numRows; ++i) {
      vector<int> temp(i);
      for (int j = 0; j < i; ++j) {
        temp[j] = get(res.back(), j - 1) + get(res.back(), j);
      }
      res.push_back(temp);
    }

    return res;
  }

  int get(const vector<int> &v, int i) {
    if (i < 0 || i >= v.size()) {
      return 0;
    }
    return v[i];
  }
};